.github/actions/setup-dfx/action.yml

```diff
@@ -0,0 +1,31 @@
+name: 'Setup DFX'
+description: Setup DFX
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/cache@v2
+      with:
+        path: |
+          ~/dfx
+          ~/.cache/dfinity
+        key: dfx-cache-0.11.1-2
+    # This step hangs on Github actions on Darwin for some reason, that
+    # is why we run this only on Linux for now
+    - name: Install DFX
+      shell: bash
+      env:
+        DFX_VERSION: 0.11.1
+      run: |
+        if [ -f ~/dfx/dfx ]
+        then
+          echo "DFX restored from cache"
+          ~/dfx/dfx --version
```

Comment:
Whether or not the cache is worth it, is very much dependent on the download speed of dfx. Sometimes there seems to be some github caching kicking in for the download, making restoring from cache slower than installation.
But I expect restoring from cache to be more consistent as the time for installation seems to vary quite a bit:
* Fast example (1s download): https://github.com/dfinity/internet-identity/runs/7992688821?check_suite_focus=true
* Slow(er) example (35s download): https://github.com/dfinity/internet-identity/runs/7992485707?check_suite_focus=true

Restoring from cache takes about 18 seconds.

Reply:
Another reason for wanting caching here is that the dfx download regularly errors out with 503s and the download script itself doesn't have a retry option